staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -564,13 +564,17 @@ func (f *featureGate) featureSpec(key Feature) (FeatureSpec, error) {
 }
 
 func (f *featureGate) unsafeRecordQueried(key Feature) {
-	queriedFeatures := map[Feature]struct{}{}
-	for k := range f.queriedFeatures.Load().(map[Feature]struct{}) {
-		queriedFeatures[k] = struct{}{}
+	queriedFeatures := f.queriedFeatures.Load().(map[Feature]struct{})
+	if queriedFeatures == nil {
+		queriedFeatures = map[Feature]struct{}{}
 	}
```

Comment:
Map index expressions are safe on nil map values (they work the same as indexing an non-nil empty map value), so it should be fine to remove this too.